quodlibet/quodlibet/formats/_audio.py

```diff
@@ -536,24 +538,104 @@ def _role_call(self, role_tag, sub_keys=None):
 
     @property
     def lyric_filename(self):
-        """Returns the (potential) lyrics filename for this file"""
-
-        filename = self.comma("title").replace(u'/', u'')[:128] + u'.lyric'
-        sub_dir = ((self.comma("lyricist") or self.comma("artist"))
-                  .replace(u'/', u'')[:128])
-
-        if os.name == "nt":
-            # this was added at a later point. only use escape_filename here
-            # to keep the linux case the same as before
-            filename = escape_filename(filename)
-            sub_dir = escape_filename(sub_dir)
-        else:
-            filename = fsnative(filename)
-            sub_dir = fsnative(sub_dir)
-
-        path = os.path.join(
-            expanduser(fsnative(u"~/.lyrics")), sub_dir, filename)
-        return path
+        """Returns the validated, or default, lyrics filename for this
+        file. User defined '[memory] lyric_rootpaths' and
+        '[memory] lyric_filenames' matches take precedence"""
+
+        from quodlibet.pattern import ArbitraryExtensionFileFromPattern
+
+        def sanitise(sep, parts):
+            return sep.join(list(map(lambda s: s.replace(u'/', u'')[:128],
```

Comment:
How about `sep.join(part.replace(u'/', u'')[:128] for part in parts)`? We prefer comprehensions generally.

Reply:
done